src/geo/leaflet/leaflet-layer-view-factory.js

```diff
@@ -5,12 +5,37 @@ var LeafletPlainLayerView = require('./leaflet-plain-layer-view');
 var LeafletCartoDBLayerGroupView = require('./leaflet-cartodb-layer-group-view');
 var LeafletTorqueLayerView = require('./leaflet-torque-layer-view');
 var LeafletCartoDBWebglLayerGroupView = require('./leaflet-cartodb-webgl-layer-group-view');
+// Should we expose a Tangram.cartodb method for doing this?
+var CCSS = require('tangram-cartocss');
 
-var LayerGroupViewConstructor = function (layerGroupModel, mapModel, options) {
-  if (options.vector) {
-    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, mapModel);
+// Make this configurable and crate a pipe of decision based on the configuration.
```

Comment:
I'd do something that expects a configuration and generates the decision chain.

We have to know the data.

```javascript
let someConfig = { // this configuration could come from the server
  ...
  rules: [{
    pos: 1,
    body: `return meta.data.stats.total > 10e6;` // It have to return a boolean (true -> raster, false -> vector)
  },
  ...
  ]
  ```
With this we can configure this filters from a dashboard... or based on other circunstancies like for example the next one:

```javascript
let someConfig = { // this configuration could come from the server
  ...
  rules: [{
    pos: 1,
    body: `meta.stats.last.time > 10000 && meta.stats.last.total > meta.stats.total)
      // time spent for opening the last map in this computer with vector is greather than 10s
      // and the number of vertex is higher than the last one, open with raster`
      
  },
  ...
  ]
  ```